File: Algorithms/MostCommonWord/MostCommonWord.py
from collections import Counter
import re

class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        words = [word.lower() for word in re.split(r'[!?\',;. ]', paragraph)]
        words = Counter(words)
        banned.append('')
        for bword in banned:
            if bword in words:
                words.pop(bword)
        return words.most_common(1)[0][0]

File: Algorithms/MostCommonWord/MostCommonWord.cpp
class Solution {
public:
    string mostCommonWord(string paragraph, vector<string>& banned) {
        unordered_map<string, int> words;
        // clean paragraph
        for(int i=0; i<paragraph.size();){
            string s = "";
            while(i < paragraph.size() && isalpha(paragraph[i])){
                s.push_back(tolower(paragraph[i++]));
            }
            while(i < paragraph.size() && !isalpha(paragraph[i])){
                i++;
            }
            // count a word
            words[s]++;
        }
        for(auto x: banned) words[x] = 0;
        string res = "";
        int count = 0;
        // choose max count
        for(auto x: words){
            if(x.second > count){
                res = x.first;
                count = x.second;
            }
        }
        return res;
    }
};